File: src/main/java/ankol/mod/merger/merger/ConflictRecord.java
package ankol.mod.merger.merger;

import ankol.mod.merger.core.BaseTreeNode;
import lombok.Data;
import lombok.RequiredArgsConstructor;

/**
 * 冲突记录
 *
 * @author Ankol
 */
@Data
@RequiredArgsConstructor
public class ConflictRecord {
    /**
     * 冲突的文件名
     */
    private final String fileName;
    /**
     * 基础模组名称
     */
    private final String baseModName;
    /**
     * 合并模组名称
     */
    private final String mergeModName;

    /**
     * 冲突的签名
     */
    private final String signature;
    private final BaseTreeNode baseNode;
    private final BaseTreeNode modNode;
    /**
     * 用户选择
     */
    private Integer userChoice;

    public ConflictRecord(String fileName, String baseModName, String mergeModName, String signature, BaseTreeNode baseNode, BaseTreeNode modNode, Integer userChoice) {
        this.fileName = fileName;
        this.baseModName = baseModName;
        this.mergeModName = mergeModName;
        this.signature = signature;
        this.baseNode = baseNode;
        this.modNode = modNode;
        this.userChoice = userChoice;
    }
}

File: src/main/java/ankol/mod/merger/merger/ConflictRecord.kt
package ankol.mod.merger.merger

import ankol.mod.merger.constants.UserChoice
import ankol.mod.merger.core.BaseTreeNode

/**
 * 冲突记录
 *
 * @author Ankol
 */
data class ConflictRecord(
    /**
     * 冲突的文件名
     */
    val fileName: String,
    /**
     * 基础模组名称
     */
    val baseModName: String,
    /**
     * 合并模组名称
     */
    val mergeModName: String,
    /**
     * 冲突的签名
     */
    val signature: String,
    /**
     * base节点
     */
    var baseNode: BaseTreeNode,
    /**
     * MOD节点
     */
    var modNode: BaseTreeNode,

    /**
     * 用户选择
     */
    var userChoice: UserChoice? = null,
)
